hapi-fees/src/main/java/com/hedera/services/usage/schedule/ScheduleSignUsage.java

```diff
@@ -0,0 +1,63 @@
+package com.hedera.services.usage.schedule;
+
+/*
+ * ‌
+ * Hedera Services API Fees
+ * ​
+ * Copyright (C) 2018 - 2020 Hedera Hashgraph, LLC
+ * ​
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * ‍
+ */
+
+import com.hedera.services.usage.SigUsage;
+import com.hedera.services.usage.TxnUsageEstimator;
+import com.hederahashgraph.api.proto.java.FeeData;
+import com.hederahashgraph.api.proto.java.TransactionBody;
+
+import static com.hedera.services.usage.SingletonEstimatorUtils.ESTIMATOR_UTILS;
+import static com.hederahashgraph.fee.FeeBuilder.BASIC_ENTITY_ID_SIZE;
+
+public class ScheduleSignUsage extends ScheduleTxnUsage<ScheduleSignUsage> {
+
+	private ScheduleSignUsage(TransactionBody scheduleSignOp, TxnUsageEstimator usageEstimator) {
+		super(scheduleSignOp, usageEstimator);
+	}
+
+	public static ScheduleSignUsage newEstimate(TransactionBody scheduleSignOp, SigUsage sigUsage) {
+		return new ScheduleSignUsage(scheduleSignOp, estimatorFactory.get(sigUsage, scheduleSignOp, ESTIMATOR_UTILS));
+	}
+
+	@Override
+	ScheduleSignUsage self() {
+		return this;
+	}
+
+	public FeeData get() {
+		var op = this.op.getScheduleSign();
+
+		var txnBytes = BASIC_ENTITY_ID_SIZE;
+		var ramBytes = 0;
+		var scheduledTxSigs = 0;
+		if (op.hasSigMap()) {
+			txnBytes += scheduleEntitySizes.bptScheduleReprGiven(op.getSigMap());
+			ramBytes += scheduleEntitySizes.sigBytesInScheduleReprGiven(op.getSigMap());
+			scheduledTxSigs += op.getSigMap().getSigPairCount();
+		}
+		usageEstimator.addBpt(txnBytes);
+		usageEstimator.addRbs(ramBytes * this.expirationTimeSecs); // TODO: Here we are assuming the worst case for rb/s. It must be the delta if we want to absolutely precise
```

Comment:
This likely won't even be noticeable in the final fee, but a good point in the `TODO`...we could add a method `ScheduleSignUsage.givenTimeToExpiry(int remaining)` and call it from the `TxnResourceUsageEstimator` implementation.

Reply:
Yes, the problem with the `givenTimeToExpiry(int remaining)` (and actually the reason why I didn't implement it, is because we don't have that information yet in the state. I am sure that once we get to Scheduled TX expiry, we will require that information and it will be added, however until then, we cannot implement it

Reply:
Ah that makes sense, thanks.